### PR TITLE
issue#2589: added fully qualified names memory stream for C#

### PR DESCRIPTION
**Pull request by guohuang**
@jforsell is it what you wanted?

@wing328 please review.

### CODE REVIEW

File: bin/csharp-dotnet2-petstore.sh [hunk 1]
```diff
@@ -26,6 +26,6 @@ fi
 
 # if you've executed sbt assembly previously it will use that instead.
 export JAVA_OPTS="${JAVA_OPTS} -XX:MaxPermSize=256M -Xmx1024M -DloggerPath=conf/log4j.properties"
-ags="$@ generate -i modules/swagger-codegen/src/test/resources/2_0/petstore.json -l CsharpDotNet2 -o samples/client/petstore/csharp-dotnet2/SwaggerClientTest/Lib/SwaggerClient"
+ags="$@ generate -i modules/swagger-codegen/src/test/resources/2_0/petstore.yaml -l CsharpDotNet2 -o samples/client/petstore/csharp-dotnet2/SwaggerClientTest/Lib/SwaggerClient"
```
Comment (guohuang) [inline bin/csharp-dotnet2-petstore.sh]: @wing328 not sure if this is the right change, but when using the `.json`, the code does not compile

Comment (wing328) [inline bin/csharp-dotnet2-petstore.sh]: What compilation error did you get?

Comment (guohuang) [inline bin/csharp-dotnet2-petstore.sh]: screen shot 2016-04-13 at 9 01 11 pm
here is the snapshot of the error

Comment (wing328) [inline bin/csharp-dotnet2-petstore.sh]: That's for csharp.net2, right ?

It has not been updated to handle various edge cases in model/property naming

Comment (guohuang) [inline bin/csharp-dotnet2-petstore.sh]: yeah, did it update the wrong one?

Comment (wing328) [inline bin/csharp-dotnet2-petstore.sh]: It's not wrong as C# .net 2.0 needs similar fix.

The generator you also want to update is C# .net 4+: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/CSharpClientCodegen.java

Comment (wing328) [inline bin/csharp-dotnet2-petstore.sh]: You will probably need to update the abstract base class for C# (https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java#L113)

**Comment by wing328** [comment]
@guohuang looks good to me. Thanks for the contribution.

**Comment by jforsell** [comment]
@guohuang @wing328 Just tested this change and the generated classes now compile successfully for me. Thanks for the quick fix!

**Comment by wing328** [comment]
@jforsell thanks for performing a test to confirm.

FYI. @KevinGlinski has submitted https://github.com/swagger-api/swagger-codegen/pull/2595 (pending merge) to add auto-generated documentation (markdown) to C# client that you might later find useful.

**Pull request merged**